### ISSUE TITLE
Problems when scraping large number of URLs from sitemap

### ISSUE THREAD

**Issue by AlasdairGray** [HW-SWeL/BMUSE#82]
Problems scraping from MassBank ([reported here](https://github.com/BioSchemas/bioschemas-data-harvesting/blob/main/README.md#biohackathon-2021-harvest):
>MassBank: 37975 pages harvested using the static scraper (v0.5.0) on 28 October; 10234 pages failed to scrape; ~36k pages unaccounted for (sitemaps contain a bit over 86k pages); 4 pages correspond to home page, index pages, search page which do not have markup in them

**Comment by AlasdairGray** [comment]
The page that is being scraped contains an error in its JSON-LD
```json
{
"identifier": "MSJ00172",
"url": "https://massbank.eu/MassBank/RecordDisplay?id=MSJ00172",
"headline": "Leucinostatin A; ESI-ITFT; MS2; POSITIVE; [M+H]+; CE 40 %",
"name": "Leucinostatin A",
"description": "This MassBank record with Accession MSJ00172 contains the MS2 mass spectrum of 'Leucinostatin A'.",
"measurementTechnique": "mass spectrometry",
"datePublished": "2020-02-21",
"license": "https://creativecommons.org/licenses/by/4.0",
"citation": "I. Momose, T. Onodera, H. Doi, H. Adachi, M. Iijima, Y. Yamazaki, R. Sawa, Y. Kubota, M. Igarashi and M. Kawada, J. Nat. Prod. 82(5), 1120-1127 (2019). doi.org/10.1021/acs.jnatprod.8b00839.",
"comment": ["Correct chemical structure of Leucinostatin A was recently reported: Abe et. al. "Catalytic Asymmetric Total Synthesis and Stereochemical Revision of Leucinostatin A: A Modulator of Tumor-Stroma Interaction", Chemistry A European Journal, 23, 11792-11796 (2017). DOI:10.1002/chem.201703239.", "The sample was injected by direct infusion.", "This record was created by the financial support of MEXT/JSPS KAKENHI Grant Number 19HP8024 to the Mass Spectrometry Society of Japan."],
"alternateName": "Leucinostatin A",
"@context": "http://schema.org",
"@type": "Dataset"
}
```

Issue for BMUSE is that the sitemap URL is returned in the error message rather than the URL of the page that has the error.

**Comment by AlasdairGray** [comment]
Logs are different when scraping with a list of URLs and a sitemap
__URL List Log__
```
16:31:36.828 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - *************************** STARTING SCRAPE: 2021-11-01 at 16:31:36 GMT
16:31:36.829 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Default charset: UTF-8
16:31:36.884 [INFO] hwu.elixir.utils.ScraperProperties - class loader namesun.misc.Launcher$AppClassLoader
16:31:36.884 [INFO] hwu.elixir.utils.ScraperProperties - sun.net.www.protocol.jar.JarURLConnection$JarURLInputStream@cb644e
16:31:36.884 [INFO] hwu.elixir.utils.ScraperProperties - Default configuration read from jar file
16:31:36.884 [INFO] hwu.elixir.utils.ScraperProperties - scraper properties{maxLimitScrape=5, locationOfSitesFile=/Users/ajg33/tmp/scraper/urls2scrape.txt, schemaContext=https://schema.org/docs/jsonldcontext.
jsonld, dynamic=false, outputFolder=/Users/ajg33/tmp/scraper/scraperOutput, chromiumDriverLocation=/Users/ajg33/bin/chromeDriver, contextCounter=0}
16:31:36.885 [INFO] hwu.elixir.utils.ScraperProperties - Local configuration read from file localconfig.properties
16:31:36.885 [INFO] hwu.elixir.utils.ScraperProperties - scraper implementation version:  0.5.0
16:31:36.885 [INFO] hwu.elixir.utils.ScraperProperties - outputFolder:                    /data0/BMUSEData/scraped/data/2021-11-01/
16:31:36.885 [INFO] hwu.elixir.utils.ScraperProperties - chromiumDriverLocation:          /data0/BMUSEData/chromeDriver/chromedriver
16:31:36.885 [INFO] hwu.elixir.utils.ScraperProperties - locationOfSitesFile:             /data0/BMUSEData/URLList/urls2scrape.txt
16:31:36.885 [INFO] hwu.elixir.utils.ScraperProperties - contextCounter:                  52421
16:31:36.885 [INFO] hwu.elixir.utils.ScraperProperties - Max no. URLs to scrape:          70000
16:31:36.885 [INFO] hwu.elixir.utils.ScraperProperties - Schema.org context URL:          https://schema.org/docs/jsonldcontext.jsonld
16:31:36.885 [INFO] hwu.elixir.utils.ScraperProperties - Dynamic scrape (global setting): false
16:31:36.887 [INFO] hwu.elixir.utils.ChromeDriverCreator - Location of chromiun driver: /data0/BMUSEData/chromeDriver/chromedriver
16:31:36.890 [INFO] hwu.elixir.utils.ChromeDriverCreator - starting a new chrome driver
16:31:37.567 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Read 5 urls to scrape from /data0/BMUSEData/URLList/urls2scrape.txt.

16:31:37.567 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Attempting to scrape: https://massbank.eu/MassBank/RecordDisplay?id=MSJ00246
16:31:37.567 [INFO] hwu.elixir.scrape.scraper.ScraperFilteredCore - static scraping setting
16:31:38.413 [DEBUG] hwu.elixir.scrape.scraper.ScraperFilteredCore - Number of JSONLD sections: 1
16:31:39.696 [INFO] hwu.elixir.scrape.scraper.ScraperCore - https://massbank.eu/MassBank/RecordDisplay?id=MSJ00246 was successfully scraped and written to /data0/BMUSEData/scraped/data/2021-11-01/ with filena
me 52422
16:31:39.696 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Attempting to scrape: https://massbank.eu/MassBank/RecordDisplay?id=MSJ00259
16:31:39.696 [INFO] hwu.elixir.scrape.scraper.ScraperFilteredCore - static scraping setting
16:31:39.769 [DEBUG] hwu.elixir.scrape.scraper.ScraperFilteredCore - Number of JSONLD sections: 1
16:31:39.858 [INFO] hwu.elixir.scrape.scraper.ScraperCore - https://massbank.eu/MassBank/RecordDisplay?id=MSJ00259 was successfully scraped and written to /data0/BMUSEData/scraped/data/2021-11-01/ with filena
me 52423
16:31:39.858 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Attempting to scrape: https://massbank.eu/MassBank/RecordDisplay?id=MSJ00172
16:31:39.859 [INFO] hwu.elixir.scrape.scraper.ScraperFilteredCore - static scraping setting
16:31:39.924 [DEBUG] hwu.elixir.scrape.scraper.ScraperFilteredCore - Number of JSONLD sections: 1
16:31:39.924 [ERROR] hwu.elixir.scrape.scraper.examples.FileScraper - The JSON-LD could not be parsed for https://massbank.eu/MassBank/RecordDisplay?id=MSJ00172
16:31:39.925 [ERROR] hwu.elixir.scrape.scraper.ScraperCore - https://massbank.eu/MassBank/RecordDisplay?id=MSJ00172 was NOT successfully scraped.
16:31:39.925 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Attempting to scrape: https://massbank.eu/MassBank/RecordDisplay?id=MSJ00201
16:31:39.925 [INFO] hwu.elixir.scrape.scraper.ScraperFilteredCore - static scraping setting
16:31:39.979 [DEBUG] hwu.elixir.scrape.scraper.ScraperFilteredCore - Number of JSONLD sections: 1
16:31:40.049 [INFO] hwu.elixir.scrape.scraper.ScraperCore - https://massbank.eu/MassBank/RecordDisplay?id=MSJ00201 was successfully scraped and written to /data0/BMUSEData/scraped/data/2021-11-01/ with filena
me 52425
16:31:40.049 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Attempting to scrape: https://massbank.eu/MassBank/RecordDisplay?id=MSJ00025
16:31:40.049 [INFO] hwu.elixir.scrape.scraper.ScraperFilteredCore - static scraping setting
16:31:40.102 [DEBUG] hwu.elixir.scrape.scraper.ScraperFilteredCore - Number of JSONLD sections: 1
16:31:40.152 [INFO] hwu.elixir.scrape.scraper.ScraperCore - https://massbank.eu/MassBank/RecordDisplay?id=MSJ00025 was successfully scraped and written to /data0/BMUSEData/scraped/data/2021-11-01/ with filena
me 52426
16:31:40.153 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Scraping over.
16:31:40.155 [INFO] hwu.elixir.scrape.scraper.ScraperCore - Closing driver...
16:31:40.223 [INFO] hwu.elixir.scrape.scraper.ScraperCore - Driver closed.
16:31:40.223 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - *************************** ENDING SCRAPE: 2021-11-01 at 16:31:40 GMT
```

__Sitemap Log__
```
16:45:28.967 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - *************************** STARTING SCRAPE: 2021-11-01 at 16:45:28 GMT
16:45:28.968 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Default charset: UTF-8
16:45:29.024 [INFO] hwu.elixir.utils.ScraperProperties - class loader namesun.misc.Launcher$AppClassLoader
16:45:29.024 [INFO] hwu.elixir.utils.ScraperProperties - sun.net.www.protocol.jar.JarURLConnection$JarURLInputStream@cb644e
16:45:29.024 [INFO] hwu.elixir.utils.ScraperProperties - Default configuration read from jar file
16:45:29.024 [INFO] hwu.elixir.utils.ScraperProperties - scraper properties{maxLimitScrape=5, locationOfSitesFile=/Users/ajg33/tmp/scraper/urls2scrape.txt, schemaContext=https://schema.org/docs/jsonldcontext.jsonld, dynamic=false, outputFolder=/Users/ajg33/tmp/scraper/scraperOutput, chromiumDriverLocation=/Users/ajg33/bin/chromeDriver, contextCounter=0}
16:45:29.024 [INFO] hwu.elixir.utils.ScraperProperties - Local configuration read from file localconfig.properties
16:45:29.024 [INFO] hwu.elixir.utils.ScraperProperties - scraper implementation version:  0.5.0
16:45:29.025 [INFO] hwu.elixir.utils.ScraperProperties - outputFolder:                    /data0/BMUSEData/scraped/data/2021-11-01/
16:45:29.025 [INFO] hwu.elixir.utils.ScraperProperties - chromiumDriverLocation:          /data0/BMUSEData/chromeDriver/chromedriver
16:45:29.025 [INFO] hwu.elixir.utils.ScraperProperties - locationOfSitesFile:             /data0/BMUSEData/URLList/urls2scrape.txt
16:45:29.025 [INFO] hwu.elixir.utils.ScraperProperties - contextCounter:                  52426
16:45:29.025 [INFO] hwu.elixir.utils.ScraperProperties - Max no. URLs to scrape:          70000
16:45:29.025 [INFO] hwu.elixir.utils.ScraperProperties - Schema.org context URL:          https://schema.org/docs/jsonldcontext.jsonld
16:45:29.025 [INFO] hwu.elixir.utils.ScraperProperties - Dynamic scrape (global setting): false
16:45:29.027 [INFO] hwu.elixir.utils.ChromeDriverCreator - Location of chromiun driver: /data0/BMUSEData/chromeDriver/chromedriver
16:45:29.029 [INFO] hwu.elixir.utils.ChromeDriverCreator - starting a new chrome driver
Starting ChromeDriver 95.0.4638.54 (d31a821ec901f68d0d34ccdbaea45b4c86ce543e-refs/branch-heads/4638@{#871}) on port 11507
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
Nov 01, 2021 4:45:29 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
16:45:29.698 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Read 1 urls to scrape from /data0/BMUSEData/URLList/urls2scrape.txt.

16:45:29.699 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Attempting to scrape: https://swel.macs.hw.ac.uk/sitemap.xml
16:45:29.740 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - parse sitemap list
16:45:29.924 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Sitemap found in URL: https://swel.macs.hw.ac.uk/sitemap.xml
16:45:29.924 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Attempting to scrape: https://massbank.eu/MassBank/RecordDisplay?id=MSJ00246
16:45:29.924 [INFO] hwu.elixir.scrape.scraper.ScraperFilteredCore - static scraping setting
16:45:30.243 [DEBUG] hwu.elixir.scrape.scraper.ScraperFilteredCore - Number of JSONLD sections: 1
16:45:31.307 [INFO] hwu.elixir.scrape.scraper.ScraperCore - https://massbank.eu/MassBank/RecordDisplay?id=MSJ00246 was successfully scraped and written to /data0/BMUSEData/scraped/data/2021-11-01/ with filename 52427
16:45:31.307 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Attempting to scrape: https://massbank.eu/MassBank/RecordDisplay?id=MSJ00172
16:45:31.307 [INFO] hwu.elixir.scrape.scraper.ScraperFilteredCore - static scraping setting
16:45:31.371 [DEBUG] hwu.elixir.scrape.scraper.ScraperFilteredCore - Number of JSONLD sections: 1
16:45:31.372 [ERROR] hwu.elixir.scrape.scraper.examples.FileScraper - The JSON-LD could not be parsed for https://swel.macs.hw.ac.uk/sitemap.xml
16:45:31.372 [INFO] hwu.elixir.scrape.scraper.ScraperCore - https://massbank.eu/MassBank/RecordDisplay?id=MSJ00172 was successfully scraped and written to /data0/BMUSEData/scraped/data/2021-11-01/ with filename 52428
16:45:31.372 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Attempting to scrape: https://massbank.eu/MassBank/RecordDisplay?id=MSJ00025
16:45:31.372 [INFO] hwu.elixir.scrape.scraper.ScraperFilteredCore - static scraping setting
16:45:31.425 [DEBUG] hwu.elixir.scrape.scraper.ScraperFilteredCore - Number of JSONLD sections: 1
16:45:31.496 [INFO] hwu.elixir.scrape.scraper.ScraperCore - https://massbank.eu/MassBank/RecordDisplay?id=MSJ00025 was successfully scraped and written to /data0/BMUSEData/scraped/data/2021-11-01/ with filename 52429
16:45:31.497 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - Scraping over.
16:45:31.498 [INFO] hwu.elixir.scrape.scraper.ScraperCore - Closing driver...
16:45:31.567 [INFO] hwu.elixir.scrape.scraper.ScraperCore - Driver closed.
16:45:31.568 [INFO] hwu.elixir.scrape.scraper.examples.FileScraper - *************************** ENDING SCRAPE: 2021-11-01 at 16:45:31 GMT
```

**Comment by AlasdairGray** [comment]
Issue is related to a memory limit on JSoup.

Fixed in 08cd5a4950dc69f045465e3a25cf2d361bdd1e96
